City-of-Helsinki/drupal-hdbt-admin
# UHF-4235 Whitelist internal links

# Whitelist internal links [UHF-4235](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4235)

Check full description and instructions from https://github.com/City-of-Helsinki/drupal-hdbt/pull/252

## What was done
* Added consent checkbox for CKeditor links and Link target field widgets (Link fields), when trying to apply the `Opens in new tab` functionality.
* Removed `hdbt-icon--link-external` class from CKeditor link buttons.
* Removed redundant preprocess for Site settings default icon
   * Fixed Site settings default icon visibility 
* Refactored CKEditor link plugin
   * Fixed logic when switching attributes for the link
   * Sanitised unneeded false values from the plugin
   * Fixed design related icon issues when the default style is used.
   * Use Helfi api base InternalDomainResolver to parse and resolve the CKEditor links.
* Select2 Icon library changes
   * Override Select2 attached library as attaching the original library will create race condition if it's included in entity form and dialog (iframe) on the same page. 
   * Fixed Select2 icon disappearing icons when adding new content and loading icons in a dialog iframe.
   * Fixed library inclusion for design selection and paragraph selection and propagating event in imagePreviewer.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the HDBT Admin theme
    * `composer require drupal/hdbt_admin:dev-UHF-4235_whitelist_internal_links`
* Run `make shell`
   * In shell run: `drush locale:check;drush locale:update;drush cr`   

## How to test
* Go to edit any page with ckeditor and add a link via the link button
   * In link dialog, select the `Open in new tab` and check that another checkbox with consent appears underneath
   * Check that also f.e. Banner paragraph and List of links paragraph has the same functionality

## Other PRs related to the task
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/252
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/263
* https://github.com/City-of-Helsinki/drupal-module-helfi-api-base/pull/56
